test/test_python_dispatch.py

```diff
@@ -0,0 +1,139 @@
+import torch
+from torch.testing._internal.common_utils import TestCase, run_tests
+from torch._C import _disabled_torch_function_impl
+
+from typing import Iterator, List, Dict
+import logging
+import contextlib
+
+# TODO: TensorBase should work
+class LoggingTensor(torch.Tensor):
+    elem: torch.Tensor
+
+    @staticmethod
+    def __new__(cls, elem, *args, **kwargs):
+        r = torch.Tensor._make_subclass(cls, elem.to('meta'), elem.requires_grad)
```

Comment:
Is the idea here that the "wrapping" Tensor is a meta Tensor so that it doesn't hold any memory. And the actual Tensor is `elem` that actually has memory.

Does that mean that we can use this "trick" to replace any Tensor-like object into a Tensor subclass at no extra cost?

Reply:
Yes.

Reply:
I sorta understood that but the clarification was helpful :)